src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -302,21 +302,8 @@ protected function getConstructor(array &$data, $class, array &$context, \Reflec
      *
      * @throws RuntimeException
      */
-    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes/*, $format = null*/)
+    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes, ?string $format = null)
```

Comment:
I agree with @dunglas . To me it slightly changes the semantic. It differentiates the default value from the acceptable types (in this case `null` is the default, so its ok, but still the typehint doesn't convey the acceptable types are `string|null`).

If the default value isn't `null`, it makes the difference more obvious: https://3v4l.org/mZo9I 

If I got this signature:

```php
public function foo(string $foo = null);
```

One day I might change the default value to `$foo = 'bar'`, but need to keep the ability to use `null`. So the typehint will have to change for `?string $foo = 'bar'`, otherwise I won't be able to use `null` anymore. (of course that's not something that would happen in Symfony considering the BC promise. It's just for the example).